sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample15_MockingClientTypes.md

```diff
@@ -0,0 +1,619 @@
+CreateQueueOptions options = new(queueName)
+{
+    UserMetadata = "some metadata"
+};
+
+options.AuthorizationRules.Add(new SharedAccessAuthorizationRule(
+    "allClaims",
+    new[] { AccessRights.Manage, AccessRights.Send, AccessRights.Listen }));
+
+QueueProperties createQueue = await administrationClient.CreateQueueAsync(options);
+```
+
+## Rule Creation Mocking
+
+```C# Snippet:ServiceBus_MockingRules
+Mock<ServiceBusAdministrationClient> mockAdministrationClient = new();
+Mock<Response<RuleProperties>> mockRuleResponse = new();
+
+mockAdministrationClient
+    .Setup(client => client.CreateRuleAsync(
+        It.IsAny<string>(),
+        It.IsAny<string>(),
+        It.IsAny<CreateRuleOptions>(),
+        It.IsAny<CancellationToken>()))
+    .Callback<string, string, CreateRuleOptions, CancellationToken>((topic, sub, opts, ct) =>
+    {
+        RuleProperties mockRuleProperties = ServiceBusModelFactory.RuleProperties(opts);
+
+        mockRuleResponse.Setup(r => r.Value).Returns(mockRuleProperties);
+    })
+    .ReturnsAsync(mockRuleResponse.Object);
+
+ServiceBusAdministrationClient administrationClient = mockAdministrationClient.Object;
+
+// The rest of this snippet illustrates how to create a rule using the mocked service bus
+// administration client above, this would be where application methods creating a rule would be
+// called.
+
+string ruleName = "rule";
+CreateRuleOptions options = new(ruleName)
+{
+    Filter = new CorrelationRuleFilter { Subject = "subject" }
+};
+
+string topic = "topic";
+string subscription = "subscription";
+RuleProperties createRule = await administrationClient.CreateRuleAsync(topic, subscription, options);
+```
+
+## Rule Manager Mocking
+
+```C# Snippet:ServiceBus_MockingRuleManager
+Mock<ServiceBusClient> mockClient = new();
+Mock<ServiceBusRuleManager> mockRuleManager = new();
+
+mockRuleManager
+    .Setup(rm => rm.DeleteRuleAsync(
+        It.IsAny<string>(),
+        It.IsAny<CancellationToken>()))
+    .Returns(Task.CompletedTask);
+
+mockRuleManager
+    .Setup(rm => rm.CreateRuleAsync(
+        It.IsAny<string>(),
+        It.IsAny<RuleFilter>(),
+        It.IsAny<CancellationToken>()))
+    .Returns(Task.CompletedTask);
+
+mockClient
+    .Setup(client => client.CreateRuleManager(
+        It.IsAny<string>(),
+        It.IsAny<string>()))
+    .Returns(mockRuleManager.Object);
+
+ServiceBusClient client = mockClient.Object;
+
+// The rest of this snippet illustrates how to create and use a rule manager using the mocked service
+// bus client above, this would be where application methods using a rule manager would be called.
+
+string topic = "topic";
+string subscription = "subscription";
+
+ServiceBusRuleManager ruleManager = client.CreateRuleManager(topic, subscription);
+
+await ruleManager.DeleteRuleAsync(RuleProperties.DefaultRuleName);
+await ruleManager.CreateRuleAsync("filter", new CorrelationRuleFilter { Subject = "subject" });
+```
+
+
+## Testing message handlers
```

Comment:
I'd suggest elevating this above the administration samples.   The processor tends to be used more frequently than the admin client, I believe.